app/services/clients/products_client.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const q = require('q')
```

Comment:
We're switching to using native Promises on frontend for performance. 

We ought to do the same here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

Reply:
Agree. But we'll have to look at this later as changing this is a bit `convoluted` and causes all other client changes causing bit heavy recfactoring. Prehaps should do this as a separate effort. I'll adopt it in the new node microservice

Reply:
OK